keyless/pepper/service/src/lib.rs

```diff
@@ -310,21 +315,61 @@ async fn update_account_recovery_db(input: &PepperInput) -> Result<(), Processin
                 aud: input.aud.clone(),
                 uid_key: input.uid_key.clone(),
                 uid_val: input.uid_val.clone(),
-                last_request_unix_ms: duration_since_epoch().as_millis() as u64,
+                first_request_unix_ms_minus_1q: None,
+                last_request_unix_ms: None,
+                num_requests: None,
             };
             let doc_id = entry.document_id();
-
-            // Actual DB operation using fluent API.
-            let op_result = db.fluent()
+            let now_unix_ms = duration_since_epoch().as_millis() as i64;
+
+            // The update transactions use the following strategy.
+            // 1. If not exists, create the document for the user identifier `(iss, aud, uid_key, uid_val)`.
+            //    but leave counter/time fields unspecified.
+            // 2. `num_requests += 1`, assuming the default value is 0.
+            // 3. `last_request_unix_ms = max(last_request_unix_ms, now)`, assuming the default value is 0.
+            // 4. `first_request_unix_ms = min(first_request_unix_ms, now)`, assuming the default value is +inf.
```

Comment:
Hm, maybe we do not need (4) because the DB entry itself has a creation timestamp?
Same for (3): does the DB have an update timestamp?

Reply:
we have to backfill, where a first request time of a user may be updated to a smaller value.

the default field doesn’t help with that

Reply:
I'm confused: the `first_request` is just the time the record was first created, no? 

Also, `last_request` is the time it was last updated, no?

So I was asking if the DB itself already provides these anyway, so we don't need to explicitly track them?

Reply:
firestore does have 2 default fields for every doc: time_created and time_updated.
but `time_created` is not enough. It's possible that we created the doc for Alice but earlier (before recovery DB exists) Alice may already have a pepper request. `first_request_unix_ms` should be updated to the earlier point during backfill.

Reply:
`time_updated` is supposed to do whatever `last_request_unix_ms` does, but it's not managed by us.
Let's not assume everything will work fine and we don't need to update `last_request_unix_ms` to an earlier time.

Reply:
🤷 I would just rely on `time_created` and `time_updated`, despite the fact that `time_created` will not be correct for accounts created before this feature ships.

`time_updated` could be (accidentally?) modified by a DB manager, yes. But so could the `last_request_unix_ms`, no?

Reply:
i still think we want the data we store to be correct, or easily fixed when it we find it's not correct...
after all, it's just two in64 fields, won't be too painful to maintain?